packages/contracts/src/contracts/current/protocol/AssetTransferProxy/IAssetProxy.sol

```diff
@@ -0,0 +1,38 @@
+/*
+
+  Copyright 2018 ZeroEx Intl.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+*/
+
+pragma solidity ^0.4.21;
+
+import {IAuthorizable as IAssetProxyAuthorizable} from "../../utils/Authorizable/IAuthorizable.sol";
+
+contract IAssetProxy is IAssetProxyAuthorizable {
+
+    /// @dev Transfers assets.
+    /// @param assetMetadata Byte array encoded for the respective asset proxy.
+    /// @param from Address to transfer token from.
+    /// @param to Address to transfer token to.
+    /// @param amount Amount of token to transfer.
+    /// @return Success of transfer.
+    function transferFrom(
+        bytes assetMetadata,
+        address from,
+        address to,
+        uint256 amount)
+        public
+        returns (bool success);
```

Comment:
We should also think about batch versions of these commands. We don't have to solve it in this PR though.

Reply:
And balance + allowance checks.

Reply:
I was actually just chatting with @abandeali1 about doing something like this in the next iteration. I'll make a note.